src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -501,4 +513,26 @@ public async Task<ListResponseModel<OrganizationUserBulkResponseModel>> BulkRest
         return new ListResponseModel<OrganizationUserBulkResponseModel>(result.Select(r =>
             new OrganizationUserBulkResponseModel(r.Item1.Id, r.Item2)));
     }
+
+    private async Task HandleUpdateWithSecretsManagerAccessAsync(OrganizationUser updatedOrganizationUser, Guid savingUserId, OrganizationUserUpdateRequestModel model)
+    {
+        var additionalSmSeatsRequired = await _countNewSmSeatsRequiredQuery.CountNewSmSeatsRequiredAsync(updatedOrganizationUser.OrganizationId, 1);
+        if (additionalSmSeatsRequired > 0)
+        {
+            var organization = await _organizationRepository.GetByIdAsync(updatedOrganizationUser.OrganizationId);
+            var update = new SecretsManagerSubscriptionUpdate(organization, true);
+            update.AdjustSeats(additionalSmSeatsRequired);
+            await _updateSecretsManagerSubscriptionCommand.ValidateUpdate(update);
+            await _updateOrganizationUserCommand.UpdateUserAsync(updatedOrganizationUser, savingUserId,
+                model.Collections?.Select(c => c.ToSelectionReadOnly()), model.Groups);
+            // Only autoscale (if required) after all validation has passed so that we know it's a valid request before
+            // updating Stripe
+            await _updateSecretsManagerSubscriptionCommand.UpdateSubscriptionAsync(update);
```

Comment:
This is new, but I think we need a try/catch block around this so that we revert the change to the `OrganizationUser` if the payment fails.

`ValidateUpdate` will pre-validate the update against our data, but it doesn't contact Stripe. I assume the payment could still fail if their card is expired, for example. (or a network outage)